source/reporters/ut_junit_reporter.tpb

```diff
@@ -99,36 +144,36 @@ create or replace type body ut_xunit_reporter is
           self.print_text(trim(l_suite.before_all.error_stack) || trim(chr(10) || chr(10) || l_suite.after_all.error_stack));
           self.print_text(']]>');
           self.print_text('</system-err>');
+        else
+          self.print_text('<system-err/>');
         end if;
       end if;
-
-      for i in 1 .. a_suite.items.count loop
-        if a_suite.items(i) is of(ut_test) then
-          print_test_elements(treat(a_suite.items(i) as ut_test));
-        elsif a_suite.items(i) is of(ut_logical_suite) then
-          print_suite_elements(treat(a_suite.items(i) as ut_logical_suite), a_suite_id);
-        end if;
-      end loop;
       self.print_text('</testsuite>');
     end;
   begin
     l_suite_id := 0;
-    self.print_text('<testsuites tests="' || l_tests_count || '"' || self.get_common_item_attributes(a_run) || '>');
+    self.print_text('<testsuites tests="' || l_tests_count || '"' || self.get_common_suite_attributes(a_run) || '>');
     for i in 1 .. a_run.items.count loop
       print_suite_elements(treat(a_run.items(i) as ut_logical_suite), l_suite_id);
     end loop;
     self.print_text('</testsuites>');
   end;
 
-  member function get_common_item_attributes(a_item ut_suite_item) return varchar2 is
+  member function get_common_suite_attributes(a_item ut_suite_item) return varchar2 is
   begin
-    return ' skipped="' || a_item.results_count.disabled_count
-           || '" error="' || a_item.results_count.errored_count
-           || '" failure="' || a_item.results_count.failure_count
+    return ' disabled="' || a_item.results_count.disabled_count
+           || '" errors="' || a_item.results_count.errored_count
+           || '" failures="' || a_item.results_count.failure_count
            || '" name="' || dbms_xmlgen.convert(nvl(a_item.description, a_item.name))
            || '" time="' || ut_utils.to_xml_number_format(a_item.execution_time()) || '" ';
   end;
 
+  member function get_common_test_attributes(a_item ut_suite_item) return varchar2 is
```

Comment:
maybe name it `get_common_item_attributes` and also call it at line 167 - you will avoid duplicated lines for name/time